### ISSUE TITLE
Error "Unrecognized line" when Using Specific Arguments in Functions.

### ISSUE THREAD

**Issue by alessandracruz** [42School/norminette#453]
**Describe the bug**
I am encountering a recurring problem when checking the code of my `ft_cd` function with the norminette tool. The tool returns an "Unrecognized line" error specifically on the prototype line of the function, when a second argument is used.

**Erroneous code**
```
#include "minishell.h"

void	change_directory(const char *path)
{
	if (chdir(path) != 0)
		print_error("cd: Error changing directory");
}

void	handle_home_directory(t_minishell *shell)
{
	t_envp *home_env;

	home_env = get_env_var("HOME", shell->l_envp);
	if (home_env == NULL || home_env->value == NULL)
		print_error("cd: HOME not set");
	else
		change_directory(home_env->value);
}

void	handle_oldpwd_directory(t_minishell *shell)
{
	t_envp *oldpwd_env;

	oldpwd_env = get_env_var("OLDPWD", shell->l_envp);
	if (oldpwd_env == NULL || oldpwd_env->value == NULL)
		print_error("cd: OLDPWD not set");
	else
	{
		printf("%s\n", oldpwd_env->value);
		change_directory(oldpwd_env->value);
	}
}

void	update_pwd_oldpwd(t_minishell *shell, char *old_dir)
{
	char	new_dir[MAX_PATH_LENGHT];
	if (getcwd(new_dir, sizeof(new_dir)) != NULL)
	{
		update_env_var("PWD", new_dir, shell->l_envp);
		update_env_var("OLDPWD", old_dir, shell->l_envp)
	}
}

void	ft_cd(char **args, t_minishell *shell)
{	
	char old_dir[MAX_PATH_LENGHT];
    if (getcwd(old_dir, sizeof(old_dir)) == NULL)
	{
        print_error("cd: Error retrieving current directory");
        return ;
    }
	if (args[1] == NULL || args[1][0] == '\0')
		handle_home_directory(shell);
	else if (strcmp(args[1], "-") == 0)
		handle_oldpwd_directory(shell);
	else
		change_directory(args[1]);
	update_pwd_oldpwd(shell, old_dir);
}
```

**Additional infos**
 - OS: Ubuntu 22.04 LTS
 - python3 --version: Python 3.10.12
 - norminette -v: norminette 3.3.54

**Additional context**
The norminette is returning the following error:
```
ft_cd.c: Error!
Error: Unrecognized line (56, 28) while parsing line [<IDENTIFIER=t_minishell>, <SPACE>, <MULT>, <IDENTIFIER=shell>, <RPARENTHESIS>]
```
Line 56 corresponds to the prototype of the `ft_cd` function with two arguments: an array of strings `char **args` and a pointer to the `t_minishell` structure. The error occurs regardless of the order of the arguments.

Steps to reproduce:
1. write a function with the following prototype: `void ft_cd(t_minishell *shell, char **args);`
2. Run the norminette on the `ft_cd` file.

Attempted solution:
- I rewrote the line manually to exclude the possibility of invisible characters.
- I tested with different names for the arguments.
- I reversed the order of the arguments, but the error persisted.
- I temporarily removed the second argument, and norminette didn't show the error.

Request:
I would like to understand if this behavior is expected due to some specific rule of the standard or if it could be a bug in the norminette tool. Any clarification or suggestion on how to solve or work around this problem would be very helpful.

Thanks in advance!

**Comment by NiumXp** [comment]
Hello, @alessandracruz!

The weird behavior is because you forgot a semicolon at the end of line 40:
```
void	update_pwd_oldpwd(t_minishell *shell, char *old_dir)
{
	char	new_dir[MAX_PATH_LENGHT];
	if (getcwd(new_dir, sizeof(new_dir)) != NULL)
	{
		update_env_var("PWD", new_dir, shell->l_envp);
		update_env_var("OLDPWD", old_dir, shell->l_envp)  // Right here
	}
}
```

About your request, "it's not a bug, it's a feature", when norminette checks a line that is a function call, it will only stop when a semicolon, `and`, `or` or a comma appears. So in your case, when norminette sees the comma in the prototype of `ft_cd`, it recognizes it as the end of the function call at line 40.

**Comment by alessandracruz** [comment]
Thank you very much for your attention and for quickly pointing out what the problem was. I still understand that forgetting a simple semicolon can lead to chaos!!! I was already very tired and it seems that we go blind looking for the error. I had already corrected it without knowing when I decided to run norminette with a flag that let me correct the rest and I ended up correcting the lack of semicolon without knowing that it was the lack of it that was causing the strange behavior. Thank you!
